langchain/src/callbacks/promises.ts

```diff
@@ -0,0 +1,34 @@
+import PQueueMod from "p-queue";
+
+let queue: typeof import("p-queue")["default"]["prototype"];
+
+function createQueue() {
+  const PQueue = "default" in PQueueMod ? PQueueMod.default : PQueueMod;
+  return new PQueue({
+    autoStart: true,
+    concurrency: 1,
```

Comment:
Concurrency 1 seems a little low?

Reply:
It's to prevent handlers executing out of order